### PR TITLE
Delay the textual rendering of Diagnostics as much as possible

### PR DESCRIPTION
**Pull request by dabrahams**
Will allow us to use the test annotation framework for all our tests, including the end-to-end ones.

### CODE REVIEW

File: Sources/ValCommand/DiagnosticLog.swift [hunk 1]
```diff
@@ -0,0 +1,157 @@
+import Core
+
+extension DiagnosticSet {
+
+  /// Write a textual representation of `self` into `output` using the given style.
```
Comment (kyouko-taiga) [inline Sources/ValCommand/DiagnosticLog.swift:5]:
```suggestion
  /// Writes a textual representation of `self` into `output` using the given style.
```

File: Sources/ValCommand/StandardErrorLog.swift [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
-  /// Appends `text` to the stream.
-  func write(_ text: String) {
-    FileHandle.standardError.write(Data(text.utf8))
+  /// Appends `data` to `self`.
+  public mutating func write(_ data: Data) {
+    base.write(data)
   }
+}
+
+var standardError = OutputFileHandle(.standardError)
```
Comment (kyouko-taiga) [inline Sources/ValCommand/StandardErrorLog.swift:21]: Maybe we should define this as a static variable in `OutputFileHandle` ?

File: Sources/ValCommand/StandardErrorLog.swift [hunk 3]
```diff
@@ -1,19 +1,25 @@
 import Foundation
 
-/// A wrapper type implementing `FileHandle.standardError`'s conformance to `Log`.
-struct StandardErrorLog: Log {
+struct OutputFileHandle: TextOutputStream {
```
Comment (kyouko-taiga) [inline Sources/ValCommand/StandardErrorLog.swift:3]: File should be renamed after the type for consistency.

**Review by kyouko-taiga — APPROVED** [review]
Good work. I left a couple of inline comments but LGTM.

File: Sources/ValCommand/StandardErrorLog.swift (hunk 2 above)
Comment (dabrahams) [reply Sources/ValCommand/StandardErrorLog.swift:21]: 🤷 Makes it more inconvenient to use, if more hygienic.  I'm not attached to this design, and am happy to have you change it whenever the mood strikes.

Comment (dabrahams) [reply Sources/ValCommand/StandardErrorLog.swift:21]: or if you feel strongly, I can change it now.

Comment (dabrahams) [reply Sources/ValCommand/StandardErrorLog.swift]: I started to, but it made code ugly and hard to read.

Comment (kyouko-taiga) [reply Sources/ValCommand/StandardErrorLog.swift]: Let's keep it that way then. I don't feel strongly about it.

**Pull request merged**